be/src/util/variant.cpp

```diff
@@ -595,15 +580,32 @@ StatusOr<VariantValue> VariantValue::get_object_by_key(const VariantMetadata& me
         break;                                                                                         \
     }
 
-    switch (info.id_size) {
-        SEARCH_DICT_INDEX_SZ(1);
-        SEARCH_DICT_INDEX_SZ(2);
-        SEARCH_DICT_INDEX_SZ(3);
-        SEARCH_DICT_INDEX_SZ(4);
+        switch (info.id_size) {
+            SEARCH_DICT_INDEX_SZ(1);
+            SEARCH_DICT_INDEX_SZ(2);
+            SEARCH_DICT_INDEX_SZ(3);
+            SEARCH_DICT_INDEX_SZ(4);
+        }
+
+        return Status::OK();
+    };
+
+    RETURN_IF_ERROR(search(0));
+
+    // don't find key in the dict, then return null value.
+    if (dict_indexes.empty()) {
```

Comment:
if there is no dict index found,  we don't return error, but return null value. 
remember this case is not a abnormal case.